core/src/main/kotlin/com/justai/jaicf/reactions/Reactions.kt

```diff
@@ -92,22 +96,53 @@ abstract class Reactions {
      * This method should be implemented by every particular channel-related [Reactions].
      *
      * @param text a raw text to append to the response
+     *
+     * @return [ImageReaction] for logging purposes
      */
-    abstract fun say(text: String)
+    abstract fun say(text: String): SayReaction
```

Comment:
I think `say` should remain abstract as any channel has methods to answer text (or generate speech from text). In other case methods image or buttons are abstract, as not every channel can process them.

Reply:
So, if we need default implementation for cases when channel *does not support* certain type of reaction, why so that default implementations produce side effects like logging?

Reply:
We need these default implementations not to produce side effects, but to establish shortcut methods.
If you're working with several channels, you don't need to call `say` on every reactions implementation. You can just use `reactions.say(text)` instead of
```
reactions.telegram?.say(text)
reactions.alexa?.say(text)
```

In case of buttons.. Well, yes, it may lead to something work not as expected, e.g. when logging reaction is created but `TelephonyChannel` cannot make `buttons` reaction. But it's only framework user to blame using buttons in telephony channel.